### PR TITLE
Updates to nav, sidebar, cards. Add "Evaluating MongoDB" page.

### PR DESCRIPTION
**Pull request by nlarew**
Staging Links:
[Main Landing](https://docs-mongodbcom-staging.corp.mongodb.com/landing/nick/DOCSP-1302/index.html)
[Evaluate MongoDB](https://docs-mongodbcom-staging.corp.mongodb.com/landing/nick/DOCSP-1302/evaluate/index.html)

[Jira Epic](https://jira.mongodb.org/browse/DOCS-10983)

**Note:** Before merging we need to decide on real copy for the text in the "Evaluating MongoDB" card.

### CODE REVIEW

**Comment by nlarew** [comment]
We have the option to change the Drivers section on the main landing page to a two-column layout within its block. See below:

screen shot 2017-11-03 at 3 08 18 pm

**Comment by nlarew** [comment]
@kay-kim 
The text in the bottom left card "Evaluating MongoDB" is still filler. Once we include real body text it's good to go for a straight merge.

**Comment by jdestefano-mongo** [comment]
@kay-kim Not yet. @shannonbradshaw would like to review final tweaks in the meeting today. Nick is correct, we also need to decide on the evaluate card text.

**Comment by jdestefano-mongo** [comment]
Nick lets discuss maybe pushing it as-is to a feature branch just incase we decide to revisit the evaluating page.

Otherwise, here are the changes required before merge:
- Remove "Evaulating MongoDB" page
- Revert the "Evaluate" card to "Cloud"
- Pull Stitch out of "Cloud" section below the cards and put in it's own "Backend as a Service" section
- In the nav, pull stitch out of "Cloud" and put under "Drivers" (different PR)

**Comment by nlarew** [comment]
- Broke out Stitch block into "Backend as a Service" section.
- Moved Stitch up a level from "Cloud" on the left nav and dropdown.
- Removed "Evaluate" page.

**Comment by nlarew** [comment]
- Broke out Stitch block into "Backend as a Service" section.
- Moved Stitch up a level from "Cloud" on the left nav and dropdown.
- Removed "Evaluate" page.

File: Makefile [hunk 1]
```diff
@@ -37,11 +37,12 @@ build: style.min.css
 	mkdir -p $@
 	mkdir -p $@/cloud
 	mkdir -p $@/tools
+	mkdir -p $@/evaluate
```
Comment (jdestefano-mongo) [inline Makefile]: Can we remove the evaluate line?

Comment (nlarew) [reply Makefile]: Removed!

**Pull request merged**